unit-tests/src/test/scala/java/nio/file/FilesSuite.scala

```diff
@@ -921,6 +921,114 @@ object FilesSuite extends tests.Suite {
     }
   }
 
+  private def setupFindSymlinkContext(
+      top: File,
+      soughtName: String): Tuple4[Path, Path, Path, Path] = {
+
+    // Create environment used to test both valid and invalid symlinks.
+
+    val dir = top.toPath()
+    val d1  = dir.resolve("d1")
+    Files.createDirectory(d1)
+    assert(Files.exists(d1) && Files.isDirectory(d1))
+
+    // f0 & f1 are just to give find() a bit more complicated case.
+    val f0 = d1.resolve("f0")
+    val f1 = d1.resolve("f1")
+    Files.createFile(f0)
+    Files.createFile(f1)
+
+    val d2 = dir.resolve("d2")
+    Files.createDirectory(d2)
+    assert(Files.exists(d2) && Files.isDirectory(d2))
+
+    val symlinkTarget = d2
+
+    val sought = d2.resolve(soughtName)
+    Files.createFile(sought)
+    assert(Files.exists(sought) && Files.isRegularFile(sought))
+
+    // Tricky bit here, symlink target is a directory, not a file.
+    val symlink = d1.resolve("dirSymlink")
+    Files.createSymbolicLink(symlink, symlinkTarget)
+
+    assert(Files.exists(symlink) && Files.isSymbolicLink(symlink))
+
+    (dir, d1, d2, symlink)
+  }
+
+  test("Files.find respects FileVisitOption.FOLLOW_LINKS, valid symlinks") {
+    withTemporaryDirectory { dirFile =>
+      val soughtName             = "quaesitum" // That which is sought.
+      val (dir, d1, d2, symlink) = setupFindSymlinkContext(dirFile, soughtName)
+
+      val predicate = new BiPredicate[Path, BasicFileAttributes] {
+        override def test(path: Path, attrs: BasicFileAttributes): Boolean =
+          path.getFileName.toString == soughtName
+      }
+
+      // Test good symlink when following links.
+
+      val itFollowGood =
+        Files.find(d1, 10, predicate, FileVisitOption.FOLLOW_LINKS).iterator
+
+      assert(itFollowGood.hasNext,
+             s"Should have found a Path when following symlinks")
+
+      // We want soughtName _exactly_ so do not use endsWith(),
+      // which would report true for, say, Foo${soughtName}.
+      val foundPath      = itFollowGood.next
+      val foundNameCount = foundPath.getNameCount
+      val foundName      = foundPath.getName(foundNameCount - 1).toString
```

Comment:
I think this could be `foundPath.getFileName.toString`

Reply:
Good suggestion.  Thank you. At that point there is at least one element to the Path, so I do not have to worry about
getFileName() returning NULL.  I'll implement as soon as I can steal the cycles.

Reply:
Edited as you suggested. Thank you for the detailed review.

The change allowed me to simplify by dropping a now unnecessary comment
and the foundNameCount.